File: Nov2020/24Nov/RemoveDuplicatesFromSortedList.java
/*
Problem Name: 83. Remove Duplicates from Sorted List
Problem Link: https://leetcode.com/problems/remove-duplicates-from-sorted-list/
*/
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode prev = head, cur = head;
        while(cur != null){
            int curval = cur.val;
            while(cur != null && (cur.val == curval))
                cur = cur.next;
            prev.next = cur;
            prev = cur;
        }
        return head;
    }
}

File: Nov2020/24Nov/RemoveDuplicatesFromSortedList.cpp
/*
Problem name: 83. Remove Duplicates from Sorted List
Problem link: https://leetcode.com/problems/remove-duplicates-from-sorted-list/
*/

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(head==nullptr)    return head;
        ListNode * prev, *cur,*ext;
        prev=head;
        cur=head->next;
        while(cur!=nullptr){
            if(cur->val==prev->val){
                while(cur!=nullptr && cur->val==prev->val){
                    cur=cur->next;            
                }
                prev->next=cur;
            }
            prev=prev->next;
        }
        return head;
    }
};
